cranelift/filetests/filetests/isa/x64/return-call-indirect.clif

```diff
@@ -248,6 +248,12 @@ block0:
 ;   pushq   %rbp
 ;   movq    %rsp, %rbp
 ;   subq    %rsp, $160, %rsp
+;   movq    %rsp, %rbp
+;   movq    160(%rsp), %r11
+;   movq    %r11, 0(%rsp)
+;   movq    168(%rsp), %r11
+;   movq    %r11, 8(%rsp)
+;   subq    %rsp, $160, %rsp
```

Comment:
Definitely not a blocker for landing this PR, but it would be nice if we could get this to be a single `sub $rsp, M+N` followed by `mov temp, [$rsp + OLD_RA_OFFSET]; mov [$rsp + NEW_RA_OFFSET]` and similar for the saved FP, instead of an initial sub to establish the frame followed by a second sub for tail call arguments space.

Reply:
We're experimenting with that on a different branch based on this PR, as a follow-up. Moving the stack check before the frame setup means that we can cut out even more instructions, as there's no need to move the frame pointer as well.